### PR TITLE
Add intro/outro pages

### PR DESCRIPTION
**Pull request by ebatty**
- Manually added intro/outro pages
- Includes code for initial auto generation (not incorporated into workflow as there was a manual step for intro)

### CODE REVIEW

**Comment by ebatty** [comment]
Could be waxier as I've included code in the ci folder that we don't actually have in a workflow. This is because there is one manual step for the intro files. The initial auto generation code will be helpful if we make changes so I want it somewhere

**Comment by bgalbraith** [comment]
Do we need to use notebooks for the intro and outro? Would markdown files be possible, or are we relying on some feature of the notebooks to support the embedding of videos and slides? Using notebooks creates havoc with the process_notebooks script as it has mangled all the READMEs and created student versions of the intro/outro notebooks. Alternatively, if these are supposed to be generated dynamically as part of book creation, they shouldn't be committed to avoid this problem.

Also, if you don't want the file generation to happen as part of the book building workflow, you either need to consolidate everything into the generate_book script or update the workflow to also run the generate_intro_outro script.

**Comment by ebatty** [comment]
I'm fixing the readme generation now so should be fine. I don't want file generation as part of the book building process - these will just be manual for now. We want to add intro/outro notebooks to the readme so they need to be saved and yes we use ipywidget tabs for video embedding

**Comment by ebatty** [comment]
To be clear, the reason I want to commit intro/outro files is so I can have them in the readme so the next step is fixing that

**Comment by ebatty** [comment]
Have added readme generation fixes

File: .github/workflows/notebook-pr.yaml [hunk 1]
```diff
@@ -38,6 +38,7 @@ jobs:
           python -m pip install --upgrade pip wheel
           pip install -r requirements.txt
           pip install nbconvert pillow flake8 fuzzywuzzy[speedup]
+          pip install pyyaml
```
Comment (bgalbraith) [inline .github/workflows/notebook-pr.yaml:41]: for reference, you could just add pyyaml to the end of the line above, but this is fine

File: ci/generate_intro_outro_notebooks.py [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+def main():
+
+    with open('tutorials/materials.yml') as fh:
+        materials = yaml.load(fh, Loader=yaml.FullLoader)
+
+    for m in materials:
+        print(m['day'])
+        directory = f"{m['day']}_{''.join(m['name'].split())}"
+
+        # # Generate intro video page
```
Comment (bgalbraith) [inline ci/generate_intro_outro_notebooks.py:17]: are these meant to be commented out?

**Review by bgalbraith — APPROVED** [review]
lgtm, just a couple reference comments

**Pull request merged**